Packs/Whois/Integrations/Whois/Whois.py

```diff
@@ -7234,85 +7299,54 @@ def get_whois_raw(domain, server="", previous=None, rfc3490=True, never_cut=Fals
 
 
 def get_root_server(domain):
-    ext = domain.split(".")[-1]
-    for dble in dble_ext:
-        if domain.endswith(dble):
-            ext = dble
 
-    if ext in list(tlds.keys()):
-        entry = tlds[ext]
-        try:
+    demisto.debug(f"Attempting to get root server from domain '{domain}'...")
+    try:
+        (_, tld) = domain.rsplit(".", 1)
+        for dble in dble_ext:
+            if domain.endswith(dble):
+                tld = dble
+
+        if tld in list(tlds.keys()):
+            entry = tlds[tld]
             host = entry["host"]
-        except KeyError:
-            context = ({
-                outputPaths['domain']: {
-                    'Name': domain,
-                    'Whois': {
-                        'QueryStatus': 'Failed'
-                    }
-                },
-            })
-            if SHOULD_ERROR:
-                return_error('The domain - {} - is not supported by the Whois service'.format(domain),
-                             outputs=context)
-            else:
-                return_warning('The domain - {} - is not supported by the Whois service'.format(domain),
-                               outputs=context)
-                raise WhoisWarnningException('The domain - {} - is not supported by the Whois service'.format(domain))
-        return host
+            return host
+        else:
+            raise WhoisInvalidDomain(f"Can't parse the root server from domain '{domain}'")
 
-    else:
-        raise WhoisException("No root WHOIS server found for domain.")
+    except (KeyError, TypeError, ValueError) as e:
+        demisto.error(f"Could not get root server from domain '{domain}': {e.__class__.__name__} {e}")
+        raise WhoisInvalidDomain(f"Can't parse the root server from domain '{domain}'")
```

Comment:
Seems that we are not using the SHOULD_ERROR flag?

Reply:
Right, `SHOULD_ERROR`'s only function is deciding whether the entry returned is an error vs a warning:

```python
if should_error:
     result = CommandResults(readable_output=f"No results returned for IP {ip}", entry_type=EntryType.ERROR)
else:
     result = CommandResults(readable_output=f"No results returned for IP {ip}", entry_type=EntryType.WARNING)
```

Reply:
Sounds like removing it will break BC and suddenly users who used to not fail on this will start to fail no?